crates/node/builder/src/builder/mod.rs

```diff
@@ -359,12 +360,13 @@ where
         EngineNodeLauncher: LaunchNode<
             NodeBuilderWithComponents<RethFullAdapter<DB, N>, N::ComponentsBuilder, N::AddOns>,
         >,
+        N::ComponentsBuilder: Clone,
     {
         self.node(node).launch().await
     }
 }
 
-impl<T: FullNodeTypes> WithLaunchContext<NodeBuilderWithTypes<T>> {
+impl<T: FullNodeTypes<Types = T> + NodeTypes> WithLaunchContext<NodeBuilderWithTypes<T>> {
```

Comment:
I looks like FullNodeTypes is only impl for our internal types and is just a helper type.
so I wonder if we can try to merge more stuff into one.
but then the Types = T, doesn't make much sense. having direct access to Types on FullNodeTypes is preferred because this is used in all the builder functions of the components

need to try a few things